core/server/middleware/cors.js

```diff
@@ -30,8 +31,27 @@ function getIPs() {
     return ips;
 }
 
-// origins that always match: localhost, local IPs, etc.
-whitelist = whitelist.concat(getIPs());
+function getUrls() {
+    var urls = [url.parse(config.url).hostname];
+
+    if (config.urlSSL) {
+        urls.push(url.parse(config.urlSSL).hostname);
+    }
```

Comment:
No it probably does not need to be whitelisted. Originally I thought there might be cases where it is needed, but having re-read how `apiUrl()` is generated, I don't think there are.